ReactQt/runtime/src/websocketmodule.cpp

```diff
@@ -0,0 +1,151 @@
+ *
+ */
+
+#include <QWebSocket>
+
+#include <QDebug>
+
+#include "bridge.h"
+#include "eventdispatcher.h"
+#include "websocketmodule.h"
+
+class WebSocketModulePrivate {
+public:
+    Bridge* bridge;
+    QMap<qlonglong, QWebSocket*> sockets;
+
+    void createSocketConnection(const QUrl& url, qlonglong socketId) {
+        QWebSocket* socket = new QWebSocket();
+        sockets.insert(socketId, socket);
+
+        QObject::connect(socket, &QWebSocket::connected, [=]() {
+            qDebug() << "Socket connected. SocketId:" << socketId;
+            if (bridge) {
+                bridge->eventDispatcher()->sendDeviceEvent("websocketOpen", QVariantMap{{"id", socketId}});
+            }
+        });
+
+        QObject::connect(socket, &QWebSocket::disconnected, [=]() {
+            sockets.remove(socketId);
+            socket->deleteLater();
+            qDebug() << "Socket disconnected. SocketId:" << socketId;
+            if (bridge) {
+                bridge->eventDispatcher()->sendDeviceEvent("websocketClosed",
+                                                           QVariantMap{{"id", socketId},
+                                                                       {"code", socket->closeCode()},
+                                                                       {"reason", socket->closeReason()},
+                                                                       {"clean", ""}});
+            }
+        });
+
+        QObject::connect(socket, &QWebSocket::textMessageReceived, [=](const QString& message) {
+            qDebug() << QString("Text message %1 received for SocketId").arg(message).arg(socketId);
+            if (bridge) {
+                bridge->eventDispatcher()->sendDeviceEvent(
+                    "websocketMessage", QVariantMap{{"id", socketId}, {"type", "text"}, {"data", message}});
+            }
+        });
+
+        QObject::connect(socket, &QWebSocket::binaryMessageReceived, [=](const QByteArray& message) {
+            qDebug() << QString("Binary message of size %1 received for SocketId").arg(message.size()).arg(socketId);
+            if (bridge) {
+                bridge->eventDispatcher()->sendDeviceEvent(
+                    "websocketMessage", QVariantMap{{"id", socketId}, {"type", "binary"}, {"data", message}});
+            }
+        });
+
+        QObject::connect(socket,
+                         static_cast<void (QWebSocket::*)(QAbstractSocket::SocketError)>(&QWebSocket::error),
+                         [=](QAbstractSocket::SocketError) { qDebug() << socket->errorString(); });
+
+        socket->open(url);
+    }
+};
+
+void WebSocketModule::connect(const QUrl& url,
+                              const QVariantList& protocols,
+                              const QVariantMap& options,
+                              qlonglong socketId) {
+    Q_D(WebSocketModule);
+
+    qDebug() << "WebSocketModule::connect with args: url: " << url << " socketId: " << socketId;
```

Comment:
Quite a lot of qDebug() calls here and there. Maybe we can wrap them with `#ifdef DEBUG`?

Reply:
What if to check for RCT_DEV? RCT_DEV is enabled in Debug only.

Reply:
It seems pretty a lot interesting things happen via WebSocket. Should be good to know that something happens there.